firefox-ios/Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -360,15 +360,13 @@ class TabManagerImplementation: LegacyTabManager, Notifiable, WindowSimpleTabsPr
 
         preserveTabs()
 
-        Task(priority: .high) {
-            var sessionData: Data?
-            if !tab.isFxHomeTab {
-                sessionData = await tabSessionStore.fetchTabSession(tabID: tabUUID)
-            }
-            await selectTabWithSession(tab: tab,
-                                       previous: previous,
-                                       sessionData: sessionData)
+        var sessionData: Data?
+        if !tab.isFxHomeTab {
+            sessionData = tabSessionStore.fetchTabSession(tabID: tabUUID)
         }
+        selectTabWithSession(tab: tab,
```

Comment:
Omg, love this fix!! 🔥 Saving/loading the `webView`'s `interactionState` asynchronicity was the source of several bugs.

Can you also clear the extra `didSelectedTabChange` delegate call that is currently in `selectTabWithSession` (documented as `FXIOS-9785 Note 1`)? We shouldn't need the duplicate the call anymore since we don't have to await this Task. But we should test that too just to make sure.

You could probably do a quick search for `FXIOS-9785` and clear the other comments I made as this should address everything now!

Reply:
> Can you also clear the extra didSelectedTabChange delegate call that is currently in selectTabWithSession (documented as FXIOS-9785 Note 1)?

Yeah I wondered if we should remove that immediately. I will remove it and do another round of testing and unless there's a problem I'll push that up shortly.

Reply:
Yeah if you push that up I'll definitely re-test the incident situation and try to test the white screen issue as well.

@dataports Maybe you can test on a device as well (for the white screen bug) once @mattreaganmozilla pushes up this change (reverting back to your initial fix for it).

Reply:
I've pushed that up, I can't see any issues so far with it removed 👍

Reply:
Awesome! 👍  I will test some more and approve if I don't find anything else! (I can only test sim though)